packages/core/src/transport/transport.ts

```diff
@@ -29,7 +51,32 @@ export class HttpRequest {
         reportBeaconError(e)
       }
     }
+
+    const transportIntrospection = (event: ProgressEvent) => {
+      const req = event?.currentTarget as XMLHttpRequest
+      const xhrTransportContext: TransportErrorContext = {
+        on_line: navigator.onLine,
+        size,
+        url,
+        try_beacon: tryBeacon,
+        bytes_limit: this.bytesLimit,
+        event: {
+          type: event.type,
+          is_trusted: event.isTrusted,
+          total: event.total,
+          loaded: event.loaded,
+        },
+        request: {
+          status: req.status,
+          ready_state: req.readyState,
+          response_text: req.responseText,
+        },
+      }
+      addMonitoringMessage('XHR fallback failed', xhrTransportContext)
+    }
+
     const request = new XMLHttpRequest()
+    request.addEventListener('loadend', (event) => transportIntrospection(event))
```

Comment:
We usually wrap all asynchronous functions in a `monitor`, but here it may cause more harm than good since it may lead to an infinite loop if sending the failure message keeps failing the same way.

So I think it's fine (@bcaudan wdyt?), but we might want to add a comment about it just in case

Reply:
Should not we report only the first xhr error? like for sendBeacon